File: social-network-visualizer-frontend/src/app/interface/CommunityOverview.ts
export interface SizeCount {
    communitySize: number;
    memberCount: number;
}

export interface CommunityOverview {
    avgHashtagsPerAuthor: number;
    totalTweets: number;
    totalCommunities: number;
    minSize: number;
    maxSize: number;
    avgSize: number;
    p10: number;
    p25: number;
    median: number;
    p75: number;
    p90: number;
    sizeHistogram: SizeCount[];
    avgTweetsPerAuthor: number;
    uniqueHashtags: number;
}

File: social-network-visualizer-backend/src/main/java/com/example/social_network_visualizer_backend/dto/community/CommunityOverview.java
package com.example.social_network_visualizer_backend.dto.community;

import java.util.List;

public record CommunityOverview(
        double avgHashtagsPerAuthor,
        long totalTweets,
        int totalCommunities,
        int minSize,
        int maxSize,
        double avgSize,
        double p10, double p25, double median, double p75, double p90,
        List<SizeCount> sizeHistogram,
        double avgTweetsPerAuthor,
        long uniqueHashtags
) {}
